pallets/dkg-metadata/src/lib.rs

```diff
@@ -1574,6 +1492,126 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	pub fn create_refresh_proposal(pub_key: Vec<u8>, nonce: u32) -> Proposal<T::MaxProposalLength> {
+		let uncompressed_pub_key = Self::decompress_public_key(pub_key).unwrap_or_default();
+		let (voter_merkle_root, session_length, voter_count) = Self::create_voter_set_data();
+		let proposal = RefreshProposal {
+			voter_merkle_root,
+			session_length,
+			voter_count,
+			nonce: nonce.into(),
+			pub_key: uncompressed_pub_key,
+		};
+
+		// Store the proposal in storage. We overwrite the storage always.
+		// This is to ensure that we can force rotate and re-sign successfully.
+		CurrentRefreshProposal::<T>::put(proposal.clone());
+
+		// Encode the proposal and return the unsigned proposal.
+		let bounded_proposal_data: BoundedVec<u8, T::MaxProposalLength> =
+			proposal.encode().try_into().unwrap_or_default();
+		Proposal::Unsigned { kind: ProposalKind::Refresh, data: bounded_proposal_data }
+	}
+
+	/// Creates the voter set merkle tree and auxiliary data for the `RefreshProposal`.
+	///
+	/// The validators's DKG Id (ECDSA keys) are used to create the voter set. We
+	/// hash the public keys into Ethereum compatible 20-byte addresses using the `keccak256` hash
+	/// and insert these addresses into a minimally-sized merkle tree. This allows us to use the
+	/// Ethereum origins (`msg.sender`) on EVMs to prove membership in the voter set.
+	///
+	/// The `RefreshProposal` proposal is a message containing:
+	/// - The merkle root of the ordered voter set's Ethereum addresses.
+	/// - The session length in milliseconds
+	/// - The # of proposers accumulated in the merkle root
+	/// - The nonce of the refresh proposal
+	/// - The public key of the next DKG's shared public key.
+	///
+	/// This data is intended to be used to update the voter set on other blockchains
+	/// that need a fallback mechanism when the DKG is not available or needs to be fixed or
+	/// changed.
+	fn create_voter_set_data() -> ([u8; 32], u64, u32) {
```

Comment:
Can this return type be more readable pls. Just type alias to u64/u32 will help a lot here.

Reply:
Yep will implement. Thanks for the feedback.